### ISSUE TITLE
Start 2024 budget process

### ISSUE THREAD

**Issue by pnorman** [openstreetmap/operations#930]
I'd like to move to a model where we split out our costs to the major services we run (and "other"). I also want to put in deprecation, so we get a better idea of what a service costs to run even though all of the capital expenses may have been a previous year.

Some items we should do well in advance
* [ ] request AWS credits so AWS can continue to provide a render server
  * include credits for other AWS usage in credit request, to cover all AWS costs
* [ ] review hardware list for capital replacements required
* [ ] Identify any new services which will require hardware

**Comment by pnorman** [comment]
> * review hardware list for capital replacements required

I don't see the need for any new general-purpose machines, unless we wanted additional capacity in OSUOSL. The oldest general-purpose machines are the Gen9s in AM6 and DB4, with the AM6 ones having 480GB SATA SSDs. We have enough spare capacity we should be able to add additional services that do not require a dedicated machine to AM6 or DB4. I expect to start replacing a lot of hardware in 2025-2026, based on when we did our past purchases.

We'll be fine on rendering capacity next year unless we lose sponsorship of the AWS server, which we'll verify with the credit request.

API DB servers need a bit more consideration. y/y growth is up significantly, see #931. If that continues, eddie and karm will be full in 2024Q2. If there's something we can change to save space, they'll last into 2025Q1. In either case, I'd put the cost of a used Xeon Scalable Gen2-based server at 6k-7.5k EUR ex-vat, depending on how the market moves and final specs.

@lonvia, do we need any new hardware for nominatim next year?

**Comment by lonvia** [comment]
stormfly-04 is at capacity, mostly due to the limited IO bandwidth. It would make sense to have a machine with more capacity in the US. A second machine in the US as a replacement for dulcy works as well. It does not make sense to replace stormfly or dulcy with a machine stationed in Europe. Outside-North-America is served well enough with vhagar and longma for the next year.

Please note that a significant part of traffic to Nominatim comes from AWS machines. This traffic is much more problematic than others because a large part is bulk geocoding and using multiple machines. The brunt of this is currently handled by stormfly and vhagar. I've started to do lot more strict blocking on stormfly already and will probably do so as well on vhagar soon. But it remains a game of whack-a-mole. I don't know if another AWS sponsored machine with AWS giving preference to that machine when routing is an option. I also don't know if the AWS machines even have the necessary IO bandwidth.

**Comment by pnorman** [comment]
karm and eddie were planned for replacement next year, but it was suggested they have more NVMe drives added to them. Right now they have 10 1.92 TB drives, with 3 reserved for parity/hot spares. snap-01 has 9 3.84TB drives with 2 reserved for parity/hot spares.

To bring karm and eddie up to snap-01's capacity, we'd want to go to 16 drives with 2 reserved for parity/hot spares. This is adding 6 drives. We can't get drives that are an exact match for the original drives, but the Micron 7450s we've been using are €378/drive for a cost of €2268/machine. All the DB servers would have enough storage space for their useful lifetime.

This would leave us with only one DB server at DB4 as we have now, so when we have that as the primary server we'd be relying at snap-02 and eddie at UCL.

The other option is to get used HP DL360 Gen10s with Xeon Scalable Gen2 CPUs.

**Comment by pnorman** [comment]
> stormfly-04 is at capacity, mostly due to the limited IO bandwidth

@lonvia given the upgrades due to stormfly, do we need to budget anything for a new Nominatim server for next year?

**Comment by Firefishy** [comment]
> > stormfly-04 is at capacity, mostly due to the limited IO bandwidth
> 
> @lonvia given the upgrades due to stormfly, do we need to budget anything for a new Nominatim server for next year?

We don't yet know the impact. The new controllers and required cabling should be installed around middle of next week.

**Comment by lonvia** [comment]
> @lonvia given the upgrades due to stormfly, do we need to budget anything for a new Nominatim server for next year?

IO on stormfly is the only issue right now. We are good capacity-wise for another year (assuming we are fine with keeping dulcy).

**Comment by tomhughes** [comment]
What do we need to do to finalise this before we hit the end of the year, and hence of the current budget?

**Comment by grischard** [comment]
I'm suggesting adding to the budget: Volunteer Engagement (Sysadmin Beer Fund) 3k£

Specify based on actuals + safety margin, a budget line for Travel.

Did we spend as little as 2450 on replacement parts? Should we explicitly budget for server upgrades, or is that part of this line?

The actuals for the last 12 months for Dorothea's assistance are 20 working days, 7030.70€.

**Comment by pnorman** [comment]
> I'm suggesting adding to the budget: Volunteer Engagement (Sysadmin Beer Fund) 3k£

We split this up into travel (e.g. extending everyone's SOTM trip to work for a day) and misc (training, new videoconference equipment, etc)

> Specify based on actuals + safety margin, a budget line for Travel

I changed travel to datacenter travel in the latest version.

> Did we spend as little as 2450 on replacement parts? Should we explicitly budget for server upgrades, or is that part of this line?

Replacement parts are only for parts that have failed. These are typically PDUs. Our drive failure rate is very low since we've moved to SSDs, and we've had one stick of RAM fail and one motherboard.

We have 2k allocated for upgrading components other than the DB servers, 4k for upgrading a machine in case of complete failure of the machine. We also have a specific line item for the DB server life extension upgrades, as those are planned.

This is a total of 6k for new stuff and 2k for like-for-like replacements, as well as general contingencies. If we have more than two complete machines worth of failures and unexpected upgrades, I'm okay if we have to explain to the board why we need more money if we go through all of that. In fact, if we have that many failures, we should be asking questions as to why - I think that's more than the last five years combined.

I believe our failures in modern times have been the HP SSDs, some PSUs, old HDDs, one RAM DIMM, and one motherboard. Our unplanned upgrades have been some disks and controller (stormfly mainly) and DDR3 for an older server.
